components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/mediasession/GeckoMediaSessionDelegate.kt

```diff
@@ -30,11 +44,22 @@ internal class GeckoMediaSessionDelegate(
         mediaSession: GeckoViewMediaSession,
         metaData: GeckoViewMediaSession.Metadata
     ) {
-        engineSession.notifyObservers {
-            onMediaMetadataChanged(
-                MediaSession.Metadata(metaData.title, metaData.artist, metaData.album)
-            )
+        CoroutineScope(Dispatchers.IO).launch(artWorkDispatcher) {
```

Comment:
nit: On the one hand you specifiy the IO dispatcher here but then you also create your own dispatcher here. 🤔